### PR TITLE
DAOS-14163 cart: Do not access rpc after decref.

### PR DESCRIPTION
**Pull request by ashleypittman**
Do not read the vales from RPC desc after decref.
Only log address in decref, not contents.
Change rpc addref and decref to be void functions.
  rc was only checked in test code.

Required-githooks: true

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: src/tests/ftest/cart/iv_server.c [hunk 1]
```diff
@@ -1206,6 +1199,7 @@ int iv_test_invalidate_iv(crt_rpc_t *rpc)
 
 	rc = crt_iv_invalidate(g_ivns, 0, key, 0, CRT_IV_SHORTCUT_NONE,
 			       *sync, invalidate_done, cb_info);
+	assert(rc == 0);
```
Comment (ashleypittman) [inline src/tests/ftest/cart/iv_server.c:1202]: Same here.

**Review by frostedcmos — DISMISSED** [review]

**Review by frostedcmos — APPROVED** [review]

File: src/tests/ftest/cart/iv_server.c [hunk 2]
```diff
@@ -1098,10 +1093,10 @@ iv_test_fetch_iv(crt_rpc_t *rpc)
 	input = crt_req_get(rpc);
 	assert(input != NULL);
 
-	rc = crt_req_addref(rpc);
-	assert(rc == 0);
+	crt_req_addref(rpc);
 
 	rc = crt_iv_fetch(g_ivns, 0, &input->key, 0, 0, fetch_done, rpc);
+	assert(rc == 0);
```
[the remark this answers is not in the document]
Comment (frostedcmos) [reply src/tests/ftest/cart/iv_server.c:1099]: As discussed in pm, iv_tests perform negative testing; for an invalid key fetch scenario this call will legitimately return non-0 rc. Test validity is verified by comparing resultant data with the expected on the iv_client. Similarly 1202 assert will need to be removed

[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/tests/ftest/cart/iv_server.c:1099]: Will do. I want to see if we can get the text of the assertion failure into the job output first before I fix this.

**Review by mchaarawi — APPROVED** [review]

**Review by frostedcmos — APPROVED** [review]

**Comment by ashleypittman** [comment]
Failure seems to be DAOS-14188

**Pull request merged**